binary/function.proto

```diff
@@ -19,12 +19,9 @@ message FunctionSignature {
         bool deterministic = 7;
         bool session_dependent = 8;
 
-        oneof derivation_type {
-            Type direct = 9;
-            CustomDerivationPolicy derivation_policy = 10;
-        }
+        TypeExpression output_type = 9;
```

Comment:
It might be helpful to add a flag that says whether function has default null behavior, e.g. null in any input is guaranteed to produce a null result.